network/discovery/dv5_service.go

```diff
@@ -319,21 +310,47 @@ func (dvs *DiscV5Service) PublishENR(logger *zap.Logger) {
 		return
 	}
 
-	defer atomic.StoreInt32(&dvs.publishState, publishStateReady)
-	dvs.discover(ctx, func(e PeerEvent) {
-		metricPublishEnrPings.Inc()
-		err := dvs.dv5Listener.Ping(e.Node)
-		if err != nil {
-			if err.Error() == "RPC timeout" {
-				// ignore
-				return
-			}
-			logger.Warn("could not ping node", fields.TargetNodeENR(e.Node), zap.Error(err))
-			return
+	dvs.publishMutex.Lock()
```

Comment:
I don't get why we need the mutex? if the deadline is only changed/accessed in this func/loop..

Reply:
just so there's no race condition between the read & write operations on `publishDeadline`